File: insertionSort.java
/**
* @author 	Luca Cortinovis	luca1cortinovis@gmail.com
* @version 	1.0
* @since	24/5/2021
*
* You can find me here:
* GitHub	Luca Cortinovis(NeptuneLuke)
*/

import java.util.Scanner;
import java.util.Arrays;

public class insertionSort {


	public static void main(String[] args) {
		
		Scanner keyboard = new Scanner(System.in);
		
		int myArray[] = new int[5];
		for(int i=0; i<myArray.length; i++) {
			System.out.println("Insert number: ");
			myArray[i] = keyboard.nextInt();
		}
		System.out.println();
		
		insertionSort(myArray);
		
		System.out.println(Arrays.toString(myArray));
	}

	
	public static void insertionSort(int myArray[]) {
		
		for(int i=1; i<myArray.length; i++) {
			
			int tmp = myArray[i];
			int j = i-1;
			while(j>=0 && myArray[j] > tmp) {
				myArray[j+1] = myArray[j];
				j--;
			}
			
			myArray[j+1] = tmp;
		}
		
	}
	
	
}

File: sorting-algorithms/insertion_sort.cpp
/**
* @author 	Luca Cortinovis	luca1cortinovis@gmail.com
*
* You can find me here:
* GitHub	Luca Cortinovis(NeptuneLuke)
*/

#include <iostream>
#include <vector>


using
std::vector,
std::cout,
std::cin;


void insertion_sort(vector<int>& myVector);


int main(void) {

	vector<int> myVector(5);
	int number;
	for (int i = 0; i < myVector.size(); ++i) {
		cout << "Insert number: ";
		cin >> number;
		myVector[i] = number;
		cout << "\n";
	}


	insertion_sort(myVector);

	for (int i = 0; i < myVector.size(); ++i) {
		cout << myVector[i] << "\n";
	}

	return 0;
}


void insertion_sort(vector<int>& myVector) {

	for (int i = 1; i < myVector.size(); ++i) {

		int tmp = myVector[i];
		int j = i - 1;
		while (j >= 0 && myVector[j] > tmp) {
			myVector[j + 1] = myVector[j];
			j--;
		}

		myVector[j + 1] = tmp;
	}

}